lib/workload/stateless/stacks/filemanager/docs/MOVED_OBJECTS.md

```diff
@@ -0,0 +1,47 @@
+# Tracking moved objects
+
+The filemanager tracks records from S3 events, which do not describe how objects move from one location to another. Using
+S3 events alone, it's not possible to tell whether an object that has been deleted in one place and created in another is
+the same object that has been moved, or two different objects. This is because S3 only tracks `Created` or `Deleted`
+events.
+
+To track moved objects, the filemanager stores additional information in S3 tags, that gets copied when the object
+is moved. This allows the filemanager to track how objects move and also allows it to copy attributes when an object
+is moved/copied.
+
+## Tagging process
+
+The process of tagging is:
+
+* When an object record is ingested, the filemanager queries `Created` events for tags. 
+* If the tags can be retrieved, the filemanager looks for a key called `ingest_id`. The key name can be 
+  configured in the environment variable `FILEMANAGER_INGESTER_TAG_NAME`.
+* The tag is parsed as a UUID, and stored in the `ingest_id` column of `s3_object` for that record.
+* If the tag does not exist, then a new UUID is generated, and the object is tagged on S3 by calling `PutObjectTagging`. 
+  The new tag is also stored in the `ingest_id` column.
+* The database is also queried for any records with the same `ingest_id` so that attributes can be copied to the new record.
+
+This logic is enabled by default, but it can be switched off by setting `FILEMANAGER_INGESTER_TRACK_MOVES`. The filemanager
```

Comment:
Tagging is currently part of the ingestion process, right? 
So there's a possibility that this may slow down the ingestion and may become an issue under heavy load?
Not for now, but if that should become the case, we could think of an async tagging strategy. 
Given the option to disable tagging (or tagging failing/missing for other reasons), it would be great to think of an async tagging option.

Reply:
Yeah I agree with this, it could potentially slow things down as it's done on ingestion. It's a bit tricky because the act of tagging the object conveys the information of the move - ideally this would be done as soon as possible (i.e. on ingestion). Anything async would extend the window that the object isn't tagged, meaning that the move can't be tracked. In practice this probably wouldn't make a different if the object isn't moved as soon as it's created.

There are `s3:ObjectTagging:*` events which might be good for this.

Reply:
This is always challenging and tradeoff. Let's give it a shot!

Reply:
Yes, tricky but that's a general "issue" with event bases systems: there's an inevitable delay/asynchronicity.

And I am not saying we should implement that now. An open ticket or comment in the code to keep track of it is perfectly fine.

To compensate for potential concurrency issues, the mentioned support of checksums, name matches, etc could be used... at least to some extend. All future considerations... all good for now!